src/js/components/CheckBoxGroup/CheckBoxGroup.js

```diff
@@ -0,0 +1,99 @@
+import React, { forwardRef, useContext, useMemo } from 'react';
+
+import { Box } from '../Box';
+import { CheckBox } from '../CheckBox';
+import { FormContext } from '../Form/FormContext';
+
+export const CheckBoxGroup = forwardRef(
+  (
+    {
+      checked: checkedProp,
+      disabled: disabledProp,
+      gap = 'small', // consistent with RadioButtonGroup default
+      labelKey,
+      valueKey,
+      onChange,
+      options: optionsProp,
+      name,
+      ...rest
+    },
+    ref,
+  ) => {
+    const formContext = useContext(FormContext);
+
+    // In case option is a string, normalize it to be an object
+    const options = useMemo(
+      () =>
+        optionsProp.map(option => {
+          return typeof option === 'string'
+            ? {
+                disabled: disabledProp,
+                id: option,
+                label: option,
+              }
+            : option;
+        }),
+      [optionsProp, disabledProp],
+    );
+
+    // 'checked' is an array of checked valueKeys
+    const [checked, setChecked] = formContext.useFormContext(
+      name,
+      checkedProp,
+      '',
+    );
+
+    // Logic is necessary to maintain a proper data structure for Form logic
+    const onCheckBoxChange = (event, optionChecked) => {
+      const nextChecked = checked || [];
+      const optionIndex = nextChecked.indexOf(optionChecked);
+
+      if (optionIndex < 0) nextChecked.push(optionChecked);
+      else nextChecked.splice(optionIndex, 1);
+
+      setChecked(nextChecked);
+      if (onChange) onChange(event);
+    };
+
+    return (
+      <Box ref={ref} gap={gap} {...rest}>
+        {options.map(
+          ({
+            label: labelOption,
+            disabled,
+            checked: checkedOption,
+            ...optionRest
+          }) => {
+            const label = optionRest[labelKey] || labelOption;
```

Comment:
This won't work if `labelKey` is "label" because "label" isn't in `optionRest`.

Reply:
I'm thinking, since we are saying that the options are tied to CheckBox, that we remove `labelKey` and `valueKey` and rely on the caller supplying `label` and `value` keys, like RadioButtonGroup.

Reply:
Fixed.